pyanaconda/storage/osinstall.py

```diff
@@ -0,0 +1,2161 @@
+
+        # list comprehension that builds device ancestors should not get None as a member
+        # when searching for bootloader devices
+        bootloader_devices = []
+        if self.bootloader_device is not None:
+            bootloader_devices.append(self.bootloader_device)
+
+        # biosboot is a special case
+        for device in self.devices:
+            if device.format.type == 'biosboot':
+                bootloader_devices.append(device)
+
+        # make a list of ancestors of all used devices
+        devices = list(set(a for d in list(self.mountpoints.values()) + self.swaps + bootloader_devices
+                           for a in d.ancestors))
+
+        # devices which share information with their distinct raw device
+        complementary_devices = [d for d in devices if d.raw_device is not d]
+
+        devices.sort(key=lambda d: len(d.ancestors))
+        for device in devices:
+            cls = next((c for c in ks_map if isinstance(device, c)), None)
+            if cls is None:
+                log.info("omitting ksdata: %s", device)
+                continue
+
+            class_attr, list_attr = ks_map[cls]
+
+            cls = getattr(self.ksdata, class_attr)
+            data = cls()    # all defaults
+
+            complements = [d for d in complementary_devices if d.raw_device is device]
+
+            if len(complements) > 1:
+                log.warning("omitting ksdata for %s, found too many (%d) complementary devices", device, len(complements))
+                continue
+
+            device = complements[0] if complements else device
+
+            device.populate_ksdata(data)
+
+            parent = getattr(self.ksdata, list_attr)
+            parent.dataList().append(data)
+
+        self.fsset = FSSet(self.devicetree)
+
+    def shutdown(self):
+        """ Deactivate all devices. """
+        try:
+            self.devicetree.teardown_all()
+        except Exception:  # pylint: disable=broad-except
+            log_exception_info(log.error, "failure tearing down device tree")
+
+    def reset(self, cleanup_only=False):
+        """ Reset storage configuration to reflect actual system state.
+
+            This will cancel any queued actions and rescan from scratch but not
+            clobber user-obtained information like passphrases, iscsi config, &c
+
+            :keyword cleanup_only: prepare the tree only to deactivate devices
+            :type cleanup_only: bool
+
+            See :meth:`devicetree.Devicetree.populate` for more information
+            about the cleanup_only keyword argument.
+        """
+        # save passphrases for luks devices so we don't have to reprompt
+        self.encryption_passphrase = None
+        for device in self.devices:
+            if device.format.type == "luks" and device.format.exists:
+                self.save_passphrase(device)
+
+        if self.ksdata:
+            self.config.update(self.ksdata)
```

Comment:
Here should be probably also something like:
```
self.ignored_disks = self.ksdata.ignoredisk.ignoredisk[:]
self.exclusive_disks = self.ksdata.ignoredisk.onlyuse[:]
```

Otherwise, this information is not propagated to the device tree and eventually all disks will be used for the installation.

The kickstart test failing with a message "/sdb/sentinel no longer exists":
https://github.com/rhinstaller/kickstart-tests/blob/master/ignoredisk.ks.in

Reply:
I've send you an email.